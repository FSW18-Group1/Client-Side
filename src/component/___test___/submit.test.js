import React from 'react'
import ReactDOM  from 'react-dom';
import { render, screen, cleanup } from "@testing-library/react";
import renderer from 'react-test-renderer'
import "@testing-library/jest-dom"
import Submit from '../submit'

afterEach(() => {
    cleanup();
})

test('render Sign in in login page ', () => {
    const {getByTestId} = render(<Submit command="Sign In"/>);
    expect(getByTestId('submit')).toHaveTextContent("Sign In")
})

test('render Create Account in signup page ', () => {
    const {getByTestId} = render(<Submit command="Create Account"/>);
    expect(getByTestId('submit')).toHaveTextContent("Create Account")
})

test('matches snapshot signup submit component', () => {
    const tree = renderer.create(<Submit command="Create Account"/>).toJSON();
    expect(tree).toMatchSnapshot();
})