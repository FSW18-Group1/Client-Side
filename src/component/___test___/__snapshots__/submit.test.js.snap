// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot signup submit component 1`] = `
<button
  className="mb-4 form-btn-submit"
  data-testid="submit"
  style={
    Object {
      "backgroundColor": "rgba(220, 220, 220, 0.2)",
      "border": "none",
      "color": "white",
      "fontSize": "14px",
      "padding": "12px 0",
      "width": "100%",
    }
  }
  type="submit"
>
  Create Account
</button>
`;
